### ISSUE TITLE
Treat remote server errors as miss and produce warnings

### ISSUE THREAD

**Issue by excitoon** [bazelbuild/bazel#8173]
<s>@buchgr

Now we have a strange situation with remote cache. When cache server return some error (e.g. 503), Bazel 0.22.0 and 0.23.2 just wait until some timeout happens (about a minute?):
```
190427 09:55:33.586:WT 282 [io.netty.util.internal.logging.Slf4JLogger.warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
com.google.devtools.build.lib.remote.blobstore.http.DownloadTimeoutException: Download of '/ac/1bd94a7abd4a57e6bbd97f8834b6f46574ed1b2ef9a61a7815cd58d06002c9b0' timed out. Received 0 bytes.
	at com.google.devtools.build.lib.remote.blobstore.http.HttpDownloadHandler.exceptionCaught(HttpDownloadHandler.java:151)
	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:285)
	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:264)
	at io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:256)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireExceptionCaught(CombinedChannelDuplexHandler.java:426)
	at io.netty.channel.ChannelHandlerAdapter.exceptionCaught(ChannelHandlerAdapter.java:87)
	at io.netty.channel.CombinedChannelDuplexHandler$1.fireExceptionCaught(CombinedChannelDuplexHandler.java:147)
	at io.netty.channel.ChannelInboundHandlerAdapter.exceptionCaught(ChannelInboundHandlerAdapter.java:131)
	at io.netty.channel.CombinedChannelDuplexHandler.exceptionCaught(CombinedChannelDuplexHandler.java:233)
	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:285)
	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:264)
	at io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:256)
	at io.netty.handler.timeout.ReadTimeoutHandler.readTimedOut(ReadTimeoutHandler.java:98)
	at io.netty.handler.timeout.ReadTimeoutHandler.channelIdle(ReadTimeoutHandler.java:90)
	at io.netty.handler.timeout.IdleStateHandler$ReaderIdleTimeoutTask.run(IdleStateHandler.java:494)
	at io.netty.handler.timeout.IdleStateHandler$AbstractIdleTask.run(IdleStateHandler.java:466)
	at io.netty.util.concurrent.PromiseTask$RunnableAdapter.call(PromiseTask.java:38)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:120)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	at java.base/java.lang.Thread.run(Unknown Source)
```

It would be super awesome if there would be:

- no extra delays;
- no stop of build;
- a warning in log.

</s>

**Comment by excitoon** [comment]
@buchgr Am I right, that the code is in `HttpBlobStore.java`?

**Comment by excitoon** [comment]
@buchgr Please kindly look at this fix 1abed46038c681d87e905939a075768b10800d2a

**Comment by excitoon** [comment]
Problem is not in 503 itself, but some special sequence of closing connection. I shall investigate it soon.
